src/core/git.test.ts

```diff
@@ -0,0 +1,175 @@
+    const destDir9 = `test/pretend_github_repository/1970/9xxx`;
+
+    // Act before assertions
+    beforeEach(async () => {
+        // const testCves = [
+        //     'CVE-1970-0001',
+        //     'CVE-1970-0002',
+        //     'CVE-1970-0003',
+        //     'CVE-1970-0998',
+        //     'CVE-1970-0999',
+        // ];
+        // testCves.forEach((i) =>
+        //     fs.copyFileSync(`${srcDir}/${i}.json`, `${destDir}/${i}.json`),
+        // );
+        fs.copyFileSync(`${srcDir}/CVE-1970-0999.json`, `${destDir}/CVE-1970-0996.json`);
+        fs.copyFileSync(`${srcDir}/CVE-1970-0999.json`, `${destDir}/CVE-1970-0997.json`);
+        // update existing CVE
+        fs.copyFileSync(
+            `${srcDir}/CVE-1970-0002u.json`,
+            `${destDir}/CVE-1970-0002.json`,
+        );
+        const prevDelta = {
+            published: [],
+            updated: [],
+        };
+    });
+
+    afterEach(async () => {
+        // this should not be necessary, but it seems to clean
+        //  up the git.lock file so that we don't get errors
+        //  about 2 git clients trying to access the same repository
+        //  at the same time
+        const git = new Git({ localDir });
+        await git.status();
+    });
+
+    // Teardown (cleanup) after assertions
+    afterAll(() => {
+        fs.copyFile(`${srcDir}/CVE-1970-0002.json`, `${destDir}/CVE-1970-0002.json`, Git.genericCallback);
+        FsUtils.rm(`${destDir}/CVE-1970-0996.json`);
+        FsUtils.rm(`${destDir}/CVE-1970-0997.json`);
+    });
+
+    const localDir: string = `test/pretend_github_repository`;
+
+    it(`properly builds a Git object with default values`, () => {
+        const git = new Git();
+        // console.log(`git.localDir=${git.localDir}`);
+        // console.log(`process.cwd()=${process.cwd()}`);
+        expect(git.localDir).toEqual(`${process.cwd()}/${process.env.CVES_BASE_DIRECTORY}`);
+    });
+
+
+    it(`properly builds a Git object with specified initailizers`, () => {
+        const git = new Git({ localDir });
+        // console.log(`git.localDir=${git.localDir}`);
+        expect(git.localDir).toMatch(localDir);
+    });
+
+
+    it(`properly displays git status`, async () => {
+        const git = new Git({ localDir });
+        const status = await git.status();
+        // console.log(`status=${JSON.stringify(status, null, 2)}`);
+        expect(status.not_added).toContain(`${localDir}/1970/0xxx/CVE-1970-0997.json`);
+        expect(status.modified).toContain(`${localDir}/1970/0xxx/CVE-1970-0002.json`);
+    });
+
+    // !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+    // all tests above this line do not modify git status
+    // all tests below this line MUST undo modifications to git status so each test can assume a common starting point
+    // !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
```

Comment:
For #16, can we move this batch into a new describe block and use beforeEach/afterEach?